spec/models/user_spec.rb

```diff
@@ -1,4 +1,133 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
+  subject { described_class.new }
+
+  it 'user is valid' do
+    subject.first_name = 'John'
+    subject.last_name = 'Doe'
+    subject.email = 'johndoe@gmail.com'
+    subject.username = 'jdoe'
+    subject.phone_number = '0449 774 759'
+
+    expect(subject).to be_valid
+  end
+
+  # Verifying if object attribute presence is functional
+  it 'first name is present' do
+    subject.first_name = ' '
+    expect(subject).to_not be_valid
+  end
+
+  it 'last name is present' do
+    subject.last_name = ' '
+    expect(subject).to_not be_valid
+  end
+
+  it 'username is present' do
+    subject.username = ' '
+    expect(subject).to_not be_valid
+  end
+
+  it 'email is present' do
+    subject.email = ' '
+    expect(subject).to_not be_valid
+  end
+
+  it 'phone number is present' do
+    subject.phone_number = ' '
+    expect(subject).to_not be_valid
+  end
+
+  # Verifying length of object attributes
+
+  it 'first name is not too long' do
+    subject.first_name = 'a' * 51
+    expect(subject).to_not be_valid
+  end
```

Comment:
(style) use `context` blocks with `before` hooks to set values at runtime, or redefine your subject for the spec:
```ruby
# using before hook
context "when first name is too long" do
  before { subject.first_name = '...' } 

  it { expect(subject).not_to be_valid }
end

# redefining subject
context "when first name is too long" do
  subject(:user) { build :user, first_name: "a" * 51 }
  
  it { expect(user).not_to be_valid } # prefer it { is_expected.not_to be_valid }
end
```

Reply:
(aside) if we have a fixed length for a field, it can also be set in the schema:
```ruby
t.string :first_name, limit: 50
```

Reply:
Ah brilliant I'll definitely keep this formatting in mind, thank you Matt!